spec/lib/staging/checklist_spec.rb

```diff
@@ -424,6 +424,20 @@
       expect(subject.requires_confirmation).not_to include(other_record)
     end
 
+    it "does not spider to other commits based on matches that don't satisfy the relation_filter proc" do
+      source_record
+      other_record = SourceRecord.create
+      join = TargetAssignment.create(:source_record => source_record, :target => other_record)
+      Stagehand::Staging::Commit.capture { source_record.increment!(:counter); join.touch }
```

Comment:
Ok I added a counter column to the `TargetAssignment` table so we can call `increment!` on it.